Earn Section: Update landing page

pablinos opened the pull request:
This is is the main landing page for the Earn section using the new card layout. It's still WIP and this is a skeleton version with a couple of cards, but it will form the basis of the section as a whole.

#### Testing instructions

Use Calypso live, and go to the `/earn/` you should be presented with the new card layout. Only the recurring payments button currently works.


Fixes #34517 

Some of the bits left to do are:

- [x] Update SVGs
- [x] Include the other cards and copy
- [x] Increase the default size of the main div (it looks a bit narrow)
- [x] Add the click handlers for the CTA buttons on each card, both for upgrading and to go to the appropriate section
- [ ] Add the analytics
- [x] Find the most appropriate way of not having a Learn More link on Simple Payments
- [ ] Work out the best way to detect the activation of each feature
- [ ] Provide the 3rd variation of each card when the feature is activated

pablinos committed:
Earn Section: Create new default landing page

As part of the new design for the earn section we need a new route to
display the promo cards.

pablinos committed:
First skeleton version of the earn section homepage

eliorivero committed:
Ensure the action panel cta is anchored to bottom to avoid having the cta elements at different height when two or more action panels with different content height are located side by side

eliorivero commented:
In https://github.com/Automattic/wp-calypso/pull/34896/commits/790599f82e85caec3583c2f6da0da7ff5a3a7098

Ensure the action panel cta is anchored to bottom to avoid having the cta elements at different height when two or more action panels with different content height are located side by side

### Before 
<img width="775" alt="Captura de Pantalla 2019-07-25 a la(s) 19 41 03" src="https://user-images.githubusercontent.com/1041600/61918895-92079080-af29-11e9-8bcd-c65d240b5130.png">

### After
<img width="747" alt="Captura de Pantalla 2019-07-25 a la(s) 19 41 11" src="https://user-images.githubusercontent.com/1041600/61918894-92079080-af29-11e9-99f6-dfeefadbfbce.png">

eliorivero committed:
Add the learnMoreLink prop to PromoSection and pass it to its PromoCardCta so we can set a Learn more link from the Earn home.

eliorivero commented:
In https://github.com/Automattic/wp-calypso/pull/34896/commits/5af61e73b11a9dd0e8a8853367220c7eb8dff515

Add the learnMoreLink prop to PromoSection and pass it to its PromoCardCta so we can set a Learn more link from the Earn home.

### Before
<img width="491" alt="Captura de Pantalla 2019-07-25 a la(s) 22 40 21" src="https://user-images.githubusercontent.com/1041600/61919907-ffb5bb80-af2d-11e9-97c8-5beb4b682bc0.png">

### After
<img width="495" alt="Captura de Pantalla 2019-07-25 a la(s) 22 40 09" src="https://user-images.githubusercontent.com/1041600/61919908-004e5200-af2e-11e9-9bc9-b99b5f735194.png">

eliorivero committed:
Add cards for recurring payments, store and ads. Refactor to be able to set different content based on the plan. Add new connected prop currentPlan that holds the plan currently active on the site

a8ci18n commented on client/my-sites/earn/home.tsx:
```diff
@@ -0,0 +1,162 @@
+				path: '/calypso/images/earn/woo.svg',
+			},
+			cta: {
+				text: translate( 'Upgrade to Premium Plan' ),
```
Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Upgrade to a Premium Plan!' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

pablinos committed:
Adding the new ads icon

pablinos committed:
Changing the styles on the primary card to keep the text centred

pablinos committed:
Updated card actions based on plan features

a8ci18n commented on client/my-sites/earn/home.tsx:
```diff
@@ -0,0 +1,196 @@
+			cta: {
+				feature: FEATURE_SIMPLE_PAYMENTS,
+				upgradeButton: {
+					text: translate( 'Upgrade to Premium Plan' ),
```
Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Upgrade to a Premium Plan!' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/earn/home.tsx:
```diff
@@ -0,0 +1,196 @@
+
+	const promos: PromoSectionProps = {
+		header: {
+			title: translate( 'Start earning money' ),
```
Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'Start earning money now' )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

pablinos replied:
@davemart-in Do you think we should add 'now' into the title of the header component?

davemart-in replied:
That's fine I think. 👍

pablinos committed:
Refactored to bring the feature dectection in to this component

a8ci18n commented on client/my-sites/earn/home.tsx:
```diff
@@ -0,0 +1,199 @@
+					action: supportLink,
+			  }
+			: {
+					text: translate( 'Upgrade to Premium Plan' ),
```
Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Upgrade to a Premium Plan!' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

pablinos committed:
Ensuring that currentPlan is always a string

eliorivero committed:
Update types for free plan checking. Add missing connected props.

eliorivero committed:
Earn: hide Store card for Jetpack sites

eliorivero committed:
Merge pull request #34919 from Automattic/update/earn-page-free-plan-logic

Earn: update type for free plan checking

eliorivero committed:
Merge branch 'update/earn-page' into update/earn-section-hide-store-jp

eliorivero committed:
Merge pull request #34921 from Automattic/update/earn-section-hide-store-jp

Earn: hide store card if site is Jetpack

pablinos committed:
Earn Page: Add third state for the recurring payments card

This adds a third version of the recurring payments card which is
displayed when the customer has connected a Stripe account.

pablinos committed:
Earn Page: Add third state to Ads card

Once a customer has setup WordAds on their site, we can display a third
version of the card, with the CTA being to view their dashboard.

pablinos committed:
Changed link page when WordAds hasn't been setup

eliorivero committed:
Remove stray semicolon

eliorivero committed:
Merge pull request #34946 from Automattic/update/earn-page-ads

Earn Page: Add a third state to the ads card

eliorivero committed:
Earn: get always a boolean value instead of null. This is so whether the user never attempted to connect the Stripe account or they started and never finished, we don't get a null in the first case and a false in the second, thus causing the wrong texts to show up

pablinos committed:
Earn Page: Fix the test for if a site is on a Free Plan

We were testing the plan slug to see if the site's plan was free, but
this meant that other free plans (on Jetpack) looked lik paid plans.
This updates the logic to use the `isCurrentPlanPaid` selector.

eliorivero committed:
Merge pull request #34967 from Automattic/fix/earn-has-connected-account

Earn: get always a boolean value for connected payments account

eliorivero committed:
Merge pull request #34972 from Automattic/fix/earn-page-free-plan

Earn Page: Fix the test for a site being on a Free Plan

eliorivero committed:
Earn: track clicks on upgrade buttons saving two properties: the target upgrade plan and the context, that is, the feature card in which the upgrade button was clicked

Commit:
Update breakpoint where cards are shown at 100%

Resolves issue where buttons are awkwardly moving to a new line

Commit:
Optimized SVG's & updated ads icon

See: https://cl.ly/3daefe781bff

sirreal commented on client/components/promo-section/index.tsx:
```diff
@@ -12,6 +12,7 @@ import PromoCardCta, { Props as PromoCardCtaProps } from './promo-card/cta';
 interface PromoSectionCardProps extends PromoCardProps {
 	body: string;
 	cta?: PromoCardCtaProps;
+	learnMoreLink?: PromoCardCtaProps;
```
Is there a mistake here?

```suggestion
	learnMoreLink: PromoCardCtaProps['learnMoreLink'];
```

pablinos replied:
You're right the types got a bit confused here. I've updated them, although I'm not sure if there is a better property name to use than `ctaProps`. I'd be interested to know if you think this is better.

sirreal commented on client/components/promo-section/index.tsx:
```diff
@@ -12,6 +12,7 @@ import PromoCardCta, { Props as PromoCardCtaProps } from './promo-card/cta';
 interface PromoSectionCardProps extends PromoCardProps {
 	body: string;
 	cta?: PromoCardCtaProps;
```
Is there a mistake here?

```suggestion
	cta: PromoCardCtaProps['cta'];
```

sirreal reviewed (commented):
A couple of notes on types.

eliorivero committed:
Earn: replace Store card with Referrals card

pablinos committed:
Adding MC stats and refactored to use Redux analytics actions

eliorivero committed:
Merge pull request #34988 from Automattic/update/earn-page-dave

Responsive CSS & SVG updates

eliorivero committed:
Merge pull request #34996 from Automattic/update/store-out-referrals-in

Earn: replace Store card with Referrals card

pablinos committed:
Adding link tracking for Learn More and non-upgrade buttons

eliorivero committed:
Use sentence case in title of Referrals card

eliorivero committed:
Merge pull request #34998 from Automattic/add/earn-analytics-upgrade-with-mc

Earn Page: Add MC and Tracks analytics to upgrade buttons

pablinos committed:
Updating types for the CTA components

pablinos commented:
I've updated the types after the changes we made for analytics and the suggestions from @sirreal . I'd be interested to hear whether you both (@sirreal & @eliorivero) think it's ok now.

pablinos committed:
Update text to existing translations and for Jetpack sites

Fixes #35017

eliorivero commented:
`ctaProps` could be named `actions`, to avoid having a prop for something already named like that, for example:

```
ctaProps: {
	cta,
},
```

eliorivero committed:
Earn: update logic to get rid of let sentences and condition checking.

pablinos committed:
Earn Page: Update analytics to cover all clicks

This change updates the names of the stats to ensure they comply with
Tracks naming conventions, and so they all use the same MC stat. It also
adds additional stats on the CTA Buttons and distinguishes between those
clicks and clicks on the Learn More links.

eliorivero committed:
Earn: rename ctaProps to actions to avoid confusing with its own prop, cta. (#35030)

eliorivero committed:
Merge pull request #35031 from Automattic/update/earn-page-analytics

Earn Page: Update analytics to cover all clicks

eliorivero committed:
Earn: add content to Referral card exclusive for Jetpack sites

eliorivero committed:
Update variable name to be consistent with naming in other places. Added jetpack or wpcom to the event name for referral to know how each one performs

pablinos committed:
Promo Section: Make external links open in a new tab

This change adds a `target="_blank"` property to any CTA action that is
defined with a URL or href, rather than a callback function. It's
assumed that these actions are external links.

eliorivero committed:
Merge pull request #35056 from Automattic/update/promo-cta-target

Promo Section: Make external links open in a new tab

eliorivero committed:
Remove unnecessary typing

eliorivero committed:
Merge pull request #35038 from Automattic/add/earn-affiliate-card-jetpack

Earn: add content to Referral card exclusively for Jetpack sites
